src/main/java/hudson/plugins/scm/koji/client/KojiBuildDownloader.java

```diff
@@ -240,6 +260,20 @@ private String composeUrl(String kojiDownloadUrl, Build build, RPM rpm, String s
         return sb.toString();
     }
 
+    private String composeSrcUrl(String kojiDownloadUrl, Build build, String suffix){
+        StringBuilder sb = new StringBuilder(255);
+        sb.append(kojiDownloadUrl);
+        if (kojiDownloadUrl.charAt(kojiDownloadUrl.length() - 1) != '/') {
+            sb.append('/');
+        }
+        sb.append(build.getName()).append('/');
+        sb.append(build.getVersion()).append('/');
+        sb.append(build.getRelease()).append('/');
+        sb.append("src/");
+        sb.append(build.getName() + '-').append(build.getVersion() + '-').append(build.getRelease() + ".src." + suffix);
```

Comment:
you're killing `StringBuilder` with concatenation. (IDEA fires an alert here, please don't ignore them, they're usually mean something)